src/fortran_routing/diffusive_v02frwk/diffusive_cnt/diffusive_cnt.f90

```diff
@@ -278,11 +278,47 @@ subroutine diffnw(dtini_g, t0_g, tfin_g, saveinterval_ev_g, dt_ql_g, dt_ub_g, dt
                 q_sk_multi=1.0
                 slope = (z(ncomp-1,j)-z(ncomp,j))/dx(ncomp-1,j)
                 if (slope .le. 0.0001) slope = 0.0001
+		oldQ(ncomp,j)= iniq(ncomp,j)
                 !oldY below takes normal depth value as a result.
                 call normal_crit_y(ncomp, j, q_sk_multi, slope, oldQ(ncomp,j), oldY(ncomp,j), temp,  oldArea(ncomp,j), temp)
             end if
         end do
         ! +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+        ! compute initial water depth, celerity, and diffusivity using iniq
+        ! +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+        q_sk_multi=1.0
+        do j=1, nlinks
+            ncomp= frnw_g(j,1)
+            do i=1, ncomp
+                oldQ(i,j) = iniq(i,j)
+		newQ(i,1,j) = iniq(i,j)
+                qp(i,1,j)= iniq(i,j)
+                if (i==ncomp) then
+                    !* normal depth
+                    slope = (z(i-1,j)-z(i,j))/dx(i-1,j)
+                    if (slope .le. 0.0001) slope = 0.0001
+                    if ((frnw_g(j,2)<0.0).and.(i==ncomp)) then
+                        !*use TW boundary water elevation data
+                        oldY(ncomp,j)=oldY(ncomp,j)
+                    else
+                        !* oldY(i,j) <- normal depth for iniq(i,j) + z(i,j). Hence, oldY is water elevation [m].
+                        call normal_crit_y(i, j, q_sk_multi, slope, oldQ(i,j), oldY(i,j), temp, temp, temp)
```

Comment:
If I'm understanding this correctly, we initially populate the depth field of oldQ with normal depth everywhere, then the mesh_diffusive_backwards function is used to update this depth estimate based on the initial flow conditions? If that is the case, we can limit the number of calls to normal_crit_y to just one.